lib/propolis/src/block/crucible.rs

```diff
@@ -61,6 +172,52 @@ impl CrucibleBackend {
         }))
     }
 
+    // Communicate to Nexus that we can remove the read only parent for
+    // the given volume id.
+    async fn remove_read_only_parent(
+        volume_id: &Uuid,
+        my_address: Ipv6Addr,
+        log: slog::Logger,
+    ) {
+        // Attempt to contact nexus, report error and return if we fail.
+        let lazy_nexus_client = {
+            match LazyNexusClient::new(log.clone(), my_address) {
+                Ok(lnc) => lnc,
+                Err(e) => {
+                    error!(log, "Failed to contact Nexus: {}", e);
+                    return;
+                }
+            }
+        };
+
+        // Notify Nexus of the state change.
+        match lazy_nexus_client.get().await {
+            Ok(client) => {
+                match client
+                    .cpapi_disk_remove_read_only_parent(&volume_id)
+                    .await
+                {
+                    Ok(_) => {
+                        info!(
+                            log,
+                            "Submitted removal for read only parent on {}",
+                            volume_id,
+                        );
+                    }
+                    Err(e) => {
+                        error!(
+                            log,
+                            "Failed removal of read only parent: {}", e,
+                        );
+                    }
+                }
+            }
+            Err(e) => {
+                error!(log, "Failed to connect to Nexus: {}", e);
+            }
```

Comment:
If the failure to resolve Nexus's address was transient, does this need to be retried?

Reply:
I'm happy to put a loop around this, but then we need to decide on a retry policy.
Try once every minute, forever?
Try once every minute for five times then give up?

It's not ideal if we can't scrub the volume, but it's also not fatal.  The next time this instance
is started we will try the scrub again.

That being said, I'm happy to take input here to improve this.

Reply:
> It's not ideal if we can't scrub the volume, but it's also not fatal. The next time this instance is started we will try the scrub again.

If this is the case (i.e., nothing will get irreparably out of sync or become unusable because Propolis failed to send a "scrub finished" notification to Nexus), then I think we can just leave this as-is (though it might be good to add a comment to this effect).

Reply:
Things have moved around a bit, make sure this still makes sense to you (comment wise)